store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverter.java

```diff
@@ -351,17 +353,27 @@ protected boolean isColumnQualifierBytesValid(final byte[] bytes, final int numP
 
     @Override
     public long buildTimestamp(final String group, final Properties properties) {
-        Long timestamp = null;
-        if (null != timestampProperty) {
+        Long timestamp;
+        if (timestampProperty != null) {
             timestamp = (Long) properties.get(timestampProperty);
+            return timestamp;
         }
 
-        if (null == timestamp) {
-            if (aggregatedGroups.contains(group)) {
-                timestamp = AccumuloStoreConstants.DEFAULT_TIMESTAMP;
+        // Check if aggregating to see what timestamp we should apply
+        if (aggregatedGroups.contains(group)) {
+            Stream<TypeDefinition> typeStream = StreamSupport.stream(
+                schema.getElement(group).getPropertyTypeDefs().spliterator(), true);
```

Comment:
Perhaps these checks can be done once when the ElementConverter is created and the results cached in a Map? I just fear that running these checks on every AddElement would be inefficient.
I suppose this would only work if the ElementConverter is created once and reused, which I am assuming it is but uncertain.

Reply:
That was my concern too, yeah I wasn't sure if the ElementConverter was created then reused, I guess if it isn't then it probably should be anyway so caching a list of all the groups that use this type of aggregator should cut some of the logic out and be more efficient.

Reply:
Having a little dig around, it seems the ElementConverter is created in a bunch of `init` functions and reused:
https://github.com/gchq/Gaffer/blob/dde023fb2f4a2c585612f3a382f4f76b1933aada/store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/impl/CoreKeyGroupByCombiner.java#L365-L373
Would also make sense as to why `timestampProperty` and `aggregatedGroups` are cached:
https://github.com/gchq/Gaffer/blob/dde023fb2f4a2c585612f3a382f4f76b1933aada/store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/key/core/AbstractCoreKeyAccumuloElementConverter.java#L60-L64

Reply:
Good to know I'll move the logic to the constructor then